### PR TITLE
Chore/show unlock icon next to view entities

### PR DESCRIPTION
**Pull request by saltcod**
— want to show icons next to views / materialized views / foreign tables if they don't meet security conditions
— show warnings / more info in GridHeaderActions based on this info

screenshot-2024-04-24-at-16 23 42

**To test:** 

1. run the countries SQL Quickstart /project/_/sql/quickstarts
2. next create a view, materialized view and a foreign table using https://gist.github.com/saltcod/17e5f9b84317d4c1618c7dfd410a8139

https://www.notion.so/supabase/Show-warning-if-there-is-a-foreign-table-materialized-view-table-in-a-schema-exposed-to-postgrest-bfb729d8a6c049f9944cb0405ad68f6e

### CODE REVIEW

File: apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx [hunk 1]
```diff
@@ -247,9 +271,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
...
+                      Learn more
+                    </Link>
+                  </Button>
+                </div>
+              </div>
+            </PopoverContent_Shadcn_>
+          </Popover_Shadcn_>
+        )}
+
+        {isForeignTable && entityType.schema === 'public' && (
```
Comment (saltcod) [inline apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx:344]: these are a little repetitive, but i converted them to a component and seemed to make it slightly worse. 

ex: these have a link as the CTA, the tables CTA has an onClick handler.

File: apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx [hunk 2]
```diff
@@ -247,9 +274,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
           )
         ) : null}
 
+        {/* need to check if view is security invoker or definer */}
+        {isView && viewIsSecurityDefiner && (
+          <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+            <PopoverTrigger_Shadcn_ asChild>
+              <Button type="warning" icon={<Unlock strokeWidth={1.5} />}>
+                View is public
```
Comment (inian) [inline apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx:282]:
```suggestion
                View is defined with the Security Definer property
```

File: apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx [hunk 3]
```diff
@@ -247,9 +274,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
...
+              </div>
+            </PopoverContent_Shadcn_>
+          </Popover_Shadcn_>
+        )}
+
+        {isMaterializedView && entityType.schema === 'public' && (
+          <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+            <PopoverTrigger_Shadcn_ asChild>
+              <Button type="warning" icon={<Unlock strokeWidth={1.5} />}>
+                View is public
```
Comment (inian) [inline apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx:319]:
```suggestion
                 View is defined with the Security Definer property
```

File: apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx [hunk 4]
```diff
@@ -247,9 +274,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
...
+              </Button>
+            </PopoverTrigger_Shadcn_>
+            <PopoverContent_Shadcn_ className="min-w-[395px] text-sm" align="end">
+              <h3 className="flex items-center gap-2">
+                <Unlock size={16} /> Secure your View
+              </h3>
+              <div className="grid gap-2 mt-4 text-foreground-light text-sm">
+                <p>
+                  Materialized view is in the public schema and can be queried by anon and
```
Comment (inian) [inline apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx:328]:
```suggestion
                  Materialized view in the public schema makes that view accessible via your project's
                  APIs.
```

To keep it consistent with the earlier message for a view

File: apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx [hunk 5]
```diff
@@ -247,9 +274,108 @@ const GridHeaderActions = ({ table, isViewSelected, isTableSelected }: GridHeade
...
+              </div>
+            </PopoverContent_Shadcn_>
+          </Popover_Shadcn_>
+        )}
+
+        {isForeignTable && entityType.schema === 'public' && (
+          <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+            <PopoverTrigger_Shadcn_ asChild>
+              <Button type="warning" icon={<Unlock strokeWidth={1.5} />}>
+                Foreign table is public
```
Comment (inian) [inline apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx:351]:
```suggestion
                Foreign table is accessible via your project's APIs
```

**Review by inian — APPROVED** [review]
approved messaging

**Review by ivasilov — APPROVED** [review]

**Pull request merged**